### PR TITLE
Update landing page

### PR DESCRIPTION
**Pull request by stevhliu**
This PR updates `index.md` to focus only on the `huggingface_hub` library and removes mentions of the widgets and community API inference. This page provides: 

- A brief overview of what the Hub is and how the `huggingface_hub` is related to the Hub and what it can do. I think there may be some potential for confusion from users over the naming of the Hub (as a product) and `huggingface_hub` (as a library), so please let me know if I've made this distinction clear enough here!
- A lot of guardrails that will help users down the right path and find what they are looking for. This ensures a user can choose their topic and then find a resource to help them accomplish their task. Once we've merged the other PRs, I can add the links in :)

I didn't mention integrating third-party libraries yet since we don't really have anything concrete yet. We can update the landing page once we have something (next on my to-do list!).

### CODE REVIEW

File: docs/source/index.mdx [hunk 1]
```diff
@@ -1,76 +1,22 @@
...
+
+Keep reading to learn even more about how to:
+
+- Delete and clone a repository, and create and update a branch.
+- Download an entire repository and use regex matching to filter and download specific
+  files.
+- Pull files from a repository before adding, committing, and pushing to the Hub.
+- Search thousands of models and datasets on the Hub with specific filters and
+  parameters to only return the best results.
+- Access the Inference API for accelerated inference.
```
Comment (adrinjalali) [inline docs/source/index.mdx:22]: I think it makes sense for these to be links to those sections.

**Review by adrinjalali — COMMENTED** [review]
If this is the first page users see, it should include a link to the quick start guide, how to contribute, and code of conduct. (we should write the ones we don't have, not in this PR though). But a link to the github repo is an easy fix.

File: docs/source/index.mdx (hunk 1 above)
Comment (stevhliu) [reply docs/source/index.mdx:22]: For sure, don't know why I thought I needed to wait for #840 to merge before adding the links! 🤦🏻‍♂️

**Review by LysandreJik — APPROVED** [review]
Thank you @stevhliu

File: docs/source/index.mdx [hunk 2]
```diff
@@ -1,76 +1,38 @@
...
+  parameters to only return the best results](searching-the-hub).
+- [Access the Inference API for accelerated inference](how-to-inference).
+
+## Contribute
+
+All contributions to the `huggingface_hub` are welcomed and equally valued! 🤗 Besides
+adding or fixing existing issues in the code, you can also help improve the
+documentation by making sure it is accurate and up-to-date, help answer questions on
+issues, and request new features you think will improve the library. Take a look at the
+[contribution
```
Comment (adrinjalali) [inline docs/source/index.mdx:31]: It'd be nice to have both this contributing guide and our CoC rendered in the docs page. Not in this PR, but something to consider as a future PR.

**Review by adrinjalali — APPROVED** [review]
Thanks @stevhliu .

**Pull request merged**